### PR TITLE
Bug Fix: Complex Password Quote

### PR DESCRIPTION
**Pull request by hpractv**
# Summary of PR changes

- Changed the quotes to handle to fix bug with generated complex passwords

## PR Requirements
- [x] For JavaScript actions, the action has been recompiled.  
  - See the *Recompiling* section of the repository's README.md for more details.
  - This does not apply to Composite Run Steps actions unless a package.json is present and contains a build script.
- [x] For major, minor, or breaking changes, at least one of the commit messages contains the appropriate `+semver:` keywords.
  - See the *Incrementing the Version* section of the repository's README.md for more details.
- [x] The examples in the repository's `README.md` have been updated with the new version.  
  - See the *Incrementing the Version* section of the repository's README.md for more details on how the version will be incremented.
- [x] The action code does not contain sensitive information.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -34,8 +34,8 @@ runs:
       id: action
       shell: powershell
       run: |
-        $service_credential_password = ConvertTo-SecureString -String "${{ inputs.service-credential-password }}" -AsPlainText -Force
-        $secure_password = ConvertTo-SecureString -String "${{ inputs.service-account-password }}" -AsPlainText -Force
+        $service_credential_password = ConvertTo-SecureString -String '${{ inputs.service-credential-password }}' -AsPlainText -Force
+        $secure_password = ConvertTo-SecureString -String '${{ inputs.service-account-password }}' -AsPlainText -Force
```
Comment (mike-schenk) [inline action.yml:38]: Still won't work if the passwords contain a `'` character.

**Review by mike-schenk — CHANGES_REQUESTED** [review]
I'm still looking into alternative possibilities.

File: action.yml (hunk 1 above)
Comment (mike-schenk) [reply action.yml:38]: If we can convince the raw input values into the values of environment variables, the PowerShell script can access the values from environment variables without quoting.

```
    - name: Action
      id: action
      shell: powershell
      env:
        SERVICE_CRED_PW: inputs.service-credential-password
        SERVICE_ACCOUNT_PW: inputs.service-account-password
      run: |
        $service_credential_password = ConvertTo-SecureString -String $Env:SERVICE_CRED_PW -AsPlainText -Force
        $secure_password = ConvertTo-SecureString -String $Env:SERVICE_ACCOUNT_PW -AsPlainText -Force
```

Comment (hpractv) [reply action.yml]: I tested a version of this with `deploy-windows-files` and it still gives the same error when a quote is one of the symbol characters.  I did find a fix using `@'` to encapsulate the string.

File: action.yml [hunk 2]
```diff
@@ -34,8 +34,15 @@ runs:
...
-        $secure_password = ConvertTo-SecureString -String '${{ inputs.service-account-password }}' -AsPlainText -Force
+        $service_password_string = @'
+        ${{ inputs.service-credential-password }}
+        '@
+        $service_credential_password = ConvertTo-SecureString -String $service_password_string -AsPlainText -Force
+
+        $deployment_password_string = @'
+        ${{ inputs.service-account-password }}
+        '@
+        $secure_password = ConvertTo-SecureString -String $deployment_password_string -AsPlainText -Force
```
Comment (mike-schenk) [inline action.yml:45]: Is there something wrong with my example that used the environment variable? Its fewer lines of code and, I think, easier to read.

**Review by mike-schenk — APPROVED** [review]

File: action.yml (hunk 2 above)
Comment (hpractv) [reply action.yml:45]: It is much easier to read, but when I tested it (https://github.com/im-open/deploy-windows-files/pull/9/commits/19a2779d828bc15cab4130c5faefbb56679ce4b1) I got the same errors.

**Pull request merged**